src/coreclr/jit/emitxarch.cpp

```diff
@@ -1464,7 +1464,9 @@ bool emitter::AreFlagsSetToZeroCmp(regNumber reg, emitAttr opSize, GenCondition
         return id->idOpSize() == opSize;
     }
 
-    if ((cond.GetCode() == GenCondition::NE) || (cond.GetCode() == GenCondition::EQ))
+    if ((cond.GetCode() == GenCondition::NE) || (cond.GetCode() == GenCondition::EQ) ||
+        (cond.GetCode() == GenCondition::SLE) || (cond.GetCode() == GenCondition::SLT) ||
+        (cond.GetCode() == GenCondition::SGE) || (cond.GetCode() == GenCondition::SGT))
```

Comment:
The code below requires only the zero flag to be written, but that's not sufficient for these other conditions. I expect that's the cause of the test failures.

Reply:
Thanks for the tip!  I've added the checks for other flag bits and separated them by the condition code so that flag checks are specific to the requested condition code.